receiver/azuremonitorreceiver/mocks.go

```diff
@@ -0,0 +1,412 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
The motive of this file is for usage only in test files. 
Can we use mocks_test.go ?

If not, can we have a comment in beginning of the file. Something like: 

> // NOTE: Test-only helpers.
> // These mocks exist to simplify Azure Monitor receiver tests.
> // They are not part of the receiver’s production logic and should not
> // be imported outside test files.

Reply:
I renamed to ``mocks_test.go``, you're totally right thanks!